Sources/CvPlayer.cpp

```diff
@@ -6588,43 +6542,15 @@ int CvPlayer::countCityFeatures(FeatureTypes eFeature) const
 
 int CvPlayer::countNumBuildings(BuildingTypes eBuilding) const
 {
-	PROFILE_FUNC();
-
-	CvCity* pLoopCity;
-	int iCount;
-	int iLoop;
-
-	iCount = 0;
-
-	for (pLoopCity = firstCity(&iLoop); pLoopCity != NULL; pLoopCity = nextCity(&iLoop))
-	{
-		if (pLoopCity->getNumBuilding(eBuilding) > 0)
-		{
-			iCount += pLoopCity->getNumBuilding(eBuilding);
-		}
-	}
-
-	return iCount;
+	return algo::accumulate(cities()
+		| filtered(CvCity::fn::getNumBuilding(eBuilding) > 0)
```

Comment:
The filter is pointless unless `getNumBuildings` can be less than 0. In fact it is worse than pointless as it costs way more to check the value then branch than it does to just add 0 to something! The filter should be removed!